src/main/java/org/mariadb/jdbc/HostAddress.java

```diff
@@ -108,8 +110,12 @@ public HostAddress(String host, int port, String type) {
         List<HostAddress> arr = new ArrayList<>(tokens.length);
 
         // Allow only cluster endpoint to be given unless testing
-        if (haMode == HaMode.AURORA && (tokens.length > 1 || (tokens.length == 1 && spec.indexOf(".cluster-") < 0)) && System.getProperty("auroraFailoverTesting") == null) {
-            throw new IllegalArgumentException("Invalid Aurora connection URL, address should correspond to cluster endpoint");
+        if (haMode == HaMode.AURORA && System.getProperty("auroraFailoverTesting") == null) {
+            Pattern clusterPattern = Pattern.compile("(.+)\\.cluster-([a-z0-9]+\\.[a-z0-9\\-]+\\.rds\\.amazonaws\\.com)");
+            Matcher matcher = clusterPattern.matcher(spec);
+            if ((tokens.length > 1 || (tokens.length == 1 && !matcher.find()))) {
+                throw new IllegalArgumentException("Invalid Aurora connection URL, address should correspond to cluster endpoint");
```

Comment:
Is there going to be documentation that will inform the users that using the cluster endpoint is the best scenario? Using the cluster endpoint allows us to handle edge cases that are currently not being handled.